modulemd/v2/tests/ModulemdTests/translation_helpers.py

```diff
@@ -33,40 +36,102 @@ class TestTranslationHelpers(TestBase):
     def test_translation_catalog_from_index(self):
         # Create a Modulemd.ModuleIndex object containing the YAML stream
         index = Modulemd.ModuleIndex.new()
-        yaml_path = "../test_data/f29.yaml"
-        ret, failures = index.update_from_file(yaml_path, True)
         self.assertIsNotNone(index)
+        ret, failures = index.update_from_file(
+            "%s/modulemd/v2/tests/test_data/f29.yaml" %
+            (os.getenv('MESON_SOURCE_ROOT')), True)
+        self.assertTrue(ret)
+        self.assertTrue(len(failures) == 0)
 
         # Create a babel catalog for the Modulemd.ModuleIndex object
-        catalog = get_translation_catalog_from_index(index)
+        catalog = Utils.get_translation_catalog_from_index(
+            index, "fedora-modularity-translations")
         self.assertTrue(catalog)
-
         # Check if the catalog is internally consistent
-        _, err = catalog.check()
-        self.assertNone(err)
+        self.assertTrue(catalog.check())
 
         # Check if each translatable string is mapped to its correct location
         for msg in catalog:
             for location in msg.locations:
-                props = location.split(";")
+                (module_name, stream_name, string_type,
+                 profile_name) = Utils.split_location(location)
 
-                module_name = props[0].split(":")[0]
                 module = index.get_module(module_name)
+                self.assertIsNotNone(module)
 
-                stream_name = props[0].split(":")[1]
                 stream = module.search_streams(stream_name, 0)[0]
+                self.assertIsNotNone(stream)
 
-                string_type = props[1]
                 if string_type == "summary":
-                    assert stream.get_summary("C") == msg.id
+                    self.assertEqual(stream.get_summary("C"), msg.id)
                 elif string_type == "description":
-                    assert stream.get_description("C") == msg.id
+                    self.assertEqual(stream.get_description("C"), msg.id)
                 else:
```

Comment:
At this point, I'd suggest asserting that `string_type` is `"profile"`, and that `profile_name` has a value.

Reply:
Skip the part about checking that `profile_name` has a value. What I intended to suggest was checking that `profile` has a value below, right after calling `get_profile()`.